### ISSUE TITLE
Need WorldEdit/WorldGuard vector update

### ISSUE THREAD

**Issue by SlimeDog** [RockinChaos/ItemJoin#186]
Spigot 1.13.2 latest
ItemJoin 4.9.1-b147
worldguard-legacy 7.0.0-b1754
worldedit-bukkit 7.0.0-b3933

WorldEdit introduced Vectors. I'm guessing ItemJoin needs to recompile with the new API.

```
[08:52:30] [Server thread/ERROR]: Could not pass event PlayerMoveEvent to ItemJoin v4.9.1-SNAPSHOT
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:308) ~[spigot-1.13.2.jar-2018-11-03-0808:git-Spigot-2474d93-f0398e4]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot-1.13.2.jar-2018-11-03-0808:git-Spigot-2474d93-f0398e4]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:500) ~[spigot-1.13.2.jar-2018-11-03-0808:git-Spigot-2474d93-f0398e4]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:485) ~[spigot-1.13.2.jar-2018-11-03-0808:git-Spigot-2474d93-f0398e4]
        at net.minecraft.server.v1_13_R2.PlayerConnection.a(PlayerConnection.java:948) ~[spigot-1.13.2.jar-2018-11-03-0808:git-Spigot-2474d93-f0398e4]
        at net.minecraft.server.v1_13_R2.PacketPlayInFlying.a(SourceFile:126) ~[spigot-1.13.2.jar-2018-11-03-0808:git-Spigot-2474d93-f0398e4]
        at net.minecraft.server.v1_13_R2.PacketPlayInFlying$PacketPlayInLook.a(SourceFile:88) ~[spigot-1.13.2.jar-2018-11-03-0808:git-Spigot-2474d93-f0398e4]
        at net.minecraft.server.v1_13_R2.PlayerConnectionUtils.a(SourceFile:10) ~[spigot-1.13.2.jar-2018-11-03-0808:git-Spigot-2474d93-f0398e4]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_181]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_181]
        at net.minecraft.server.v1_13_R2.SystemUtils.a(SourceFile:199) [spigot-1.13.2.jar-2018-11-03-0808:git-Spigot-2474d93-f0398e4]
        at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:900) [spigot-1.13.2.jar-2018-11-03-0808:git-Spigot-2474d93-f0398e4]
        at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:417) [spigot-1.13.2.jar-2018-11-03-0808:git-Spigot-2474d93-f0398e4]
        at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:835) [spigot-1.13.2.jar-2018-11-03-0808:git-Spigot-2474d93-f0398e4]
        at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:733) [spigot-1.13.2.jar-2018-11-03-0808:git-Spigot-2474d93-f0398e4]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
Caused by: java.lang.NoClassDefFoundError: com/sk89q/worldedit/Vector
        at me.RockinChaos.itemjoin.listeners.giveitems.RegionEnter.getGuardSetRegions(RegionEnter.java:221) ~[?:?]
        at me.RockinChaos.itemjoin.listeners.giveitems.RegionEnter.CheckInRegion(RegionEnter.java:206) ~[?:?]
        at me.RockinChaos.itemjoin.listeners.giveitems.RegionEnter.isInRegion(RegionEnter.java:186) ~[?:?]
        at me.RockinChaos.itemjoin.listeners.giveitems.RegionEnter.giveOnRegionEnter(RegionEnter.java:42) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_181]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_181]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_181]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_181]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:304) ~[spigot-1.13.2.jar-2018-11-03-0808:git-Spigot-2474d93-f0398e4]
        ... 15 more
```

**Comment by RockinChaos** [comment]
Phew.. Man the developers of WorldGuard & WorldEdit really do not want people making their own plugins compatible on both newer and older versions of their plugins.

**Tested and seems to be resolved;** http://ci.craftationgaming.com/job/ItemJoin/153/

I have done both forward and backward compatibility testing between older 1.12 builds of WorldGuard & WorldEdit as well as newer 1.13-2 builds of WorldGuard & WorldEdit. So far, I see no problems however further testing will be required.

**Comment by SlimeDog** [comment]
As the argument goes, there is no reason to do so; older versions match older versions. :)

I will test standard features on 1.13.2. Someone else will have to verify older versions.

**Comment by RockinChaos** [comment]
This issue has been officially fixed in the release of ItemJoin v5.0.0.
You can grab it here; https://www.spigotmc.org/resources/itemjoin.12661/download?version=254085

Please read the changelog as the documentation has not been updated yet;
https://www.spigotmc.org/wiki/recent-changes
